src/main/java/uk/gov/pay/directdebit/mandate/services/MandateQueryService.java

```diff
@@ -61,4 +66,30 @@ public Mandate findById(Long mandateId) {
     public List<Mandate> findAllMandatesBySetOfStatesAndMaxCreationTime(Set<MandateState> states, ZonedDateTime cutOffTime) {
         return mandateDao.findAllMandatesBySetOfStatesAndMaxCreationTime(states, cutOffTime);
     }
+    
+    public MandateSearchResults search(MandateSearchParams params, String gatewayAccountExternalId) {
```

Comment:
I’d actually be tempted to put this in its own class rather than `MandateQueryService` (`MandateSearchService`?) so that each service is quite tightly focused. It feels like a search is distinct enough from a query (for a single mandate, up until this point) to be its own thing.

Reply:
Ok done 👍